app/javascript/components/student/Editor.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react'
+import { CodeEditor } from './editor/CodeEditor'
+import { TestRunSummary } from './editor/TestRunSummary'
+import fetch from 'isomorphic-fetch'
+
+export type Submission = {
+  testsStatus: TestRunStatus
+  uuid: string
+}
+
+export enum TestRunStatus {
+  PASS = 'pass',
+  FAIL = 'fail',
+  ERROR = 'error',
+  PENDING = 'pending',
+  OPS_ERROR = 'ops_error',
+}
+
+export function Editor({ endpoint }: { endpoint: string }) {
+  const [submission, setSubmission] = useState<Submission>()
+
+  function submit(code: string) {
+    setSubmission(undefined)
+
+    fetch(endpoint, {
```

Comment:
Consider making this cancellable if a new submission is started whilst the request still runs.

Reply:
@SleeplessByte Good shout.

@kntsoriano This should happen server-side. Can you add a `# TODO: ...` at https://github.com/exercism/v3-website/blob/master/app/commands/submission/create.rb#L34 for us to cancel any other submissions to the same user's solution, and an issue about this in the project-management repo, please.

Reply:
This should *also hebben server-side.

Reply:
I've created an issue for this. I'll add this feature in a future PR.